cirq/work/observable_settings.py

```diff
@@ -0,0 +1,186 @@
+# This file contains code modified from PyQuil.
+# https://github.com/rigetti/pyquil
+# The original code's license is:
+#
+# Copyright 2016-2019 Rigetti Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Modifications are covered under:
+#
+# Copyright 2020 The Cirq developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+from typing import Union, Iterable, Dict, TYPE_CHECKING, Tuple
+
+from cirq import ops, value, protocols
+
+if TYPE_CHECKING:
+    import cirq
+    from cirq.value.product_state import _NamedOneQubitState
+
+
+@dataclasses.dataclass(frozen=True)
+class InitObsSetting:
+    """A pair of initial state and observable.
+
+    Usually, given a circuit you want to iterate through many
+    InitObsSettings to vary the initial state preparation and output
+    observable.
```

Comment:
And this is always 1:1?

Reply:
what do you mean? It's 1 circuit to many InitObsSettings (usually). This class wraps a relationship between 1 initial state and 1 observable.

Reply:
It just felt like a rigid way of tying these objects together - that for example can results in a lot of repetition of the initial states in case all your experiments are starting from the zero state. But I guess neither ProductState or PauliString are that expensive + we can later introduce these kind of optimizations if necessary. Resolving.